### PR TITLE
Added query docs generation script

### LINKED ISSUE #2076: Add automated documentation generation for queries
### Description
Each query information is defined by using the metadata.json file, inside each query folder.
The goal of this feature is to enable automatic generation of documentation for the queries, provided in a HTML format

### PR DESCRIPTION
**Pull request by felipe-avelar**
Signed-off-by: Felipe Avelar <felipe.avelar@checkmarx.com>
Closes #2076 
**Proposed Changes**

- Created query docs generator script to be integrated to CI pipeline
I submit this contribution under Apache-2.0 license.

### CODE REVIEW

File: .github/generators/docs_generator.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+from jinja2 import Template
+import os, argparse, json
+
+severities = {'High': 'HIGH', 'Medium': 'MEDIUM', 'Low': 'LOW', 'Info': 'INFO'}
+colors = {'High': '#C00', 'Medium': '#C60', 'Low': '#CC0', 'Info': '#00C'}
+template_dict = {}
+platforms = []
+
+arg_parser = argparse.ArgumentParser(description='Create query docs')
+arg_parser.add_argument('-p', nargs=1, type=Path, dest='intput_path', help='Path to read metadata.json')
```
Comment (rogeriopeixotocx) [inline .github/generators/docs_generator.py:12]: `input_path`

File: .github/generators/docs_generator.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+arg_parser.add_argument('-o', nargs=1, type=Path, dest='output_path', help='Path to output documentation files')
+arg_parser.add_argument('-t', nargs=1, type=Path, dest='templates_path', help='Path to template files')
+arg_parser.add_argument('-f', nargs='+', type=str, dest='formats', help='Formats to be created documentation, must exists on template folder')
+parsed_args = vars(arg_parser.parse_args())
+
+parsed_args['formats'] = ['.%s' % f.lstrip('.') for f in parsed_args['formats']]
+parsed_args['output_path'][0].mkdir(parents=True, exist_ok=True)
+
+for path in parsed_args['intput_path'][0].rglob('metadata.json'):
```
Comment (rogeriopeixotocx) [inline .github/generators/docs_generator.py:21]: `input_path`

File: .github/generators/docs_generator.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    category = metadata_dict['category']
+
+    if platform not in template_dict:
+      template_dict[platform] = {}
+    if severity not in template_dict[platform]:
+      template_dict[platform][severity] = {}
+    if category not in template_dict[platform][severity]:
+      template_dict[platform][severity][category] = {}
+
+    if isinstance(metadata_dict['descriptionUrl'], str):
```
Comment (rogeriopeixotocx) [inline .github/generators/docs_generator.py:38]: if it's not a string and it's an array of strings, how does it work?

File: .github/generators/docs_generator.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      metadata_dict['descriptionUrl'] = [metadata_dict['descriptionUrl']]
+    template_dict[platform][severity][category][metadata_dict['id']] = metadata_dict
+
+for platform in template_dict:
+  for file_format in parsed_args['formats']:
+    with open(os.path.join(parsed_args['templates_path'][0], 'template' + file_format)) as template:
+      # sort to be ordered by category
```
Comment (rogeriopeixotocx) [inline .github/generators/docs_generator.py:45]: `sort by category`

File: .github/generators/docs_generator.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      data = {}
+      for severity in severities:
+        if severities[severity] in template_dict[platform]:
+          data[severity] = {}
+          for category in sorted(template_dict[platform][severities[severity]].keys()):
+            data[severity] = {**data[severity], **template_dict[platform][severities[severity]][category]}
```
Comment (rogeriopeixotocx) [inline .github/generators/docs_generator.py:51]: good use of `**` dict unpacking operator not sure if we could make it easier to read... at least the second part `template_dict[platform][severities[severity]][category]` by putting this into an intermediate variable that had a meaningful name.

File: .github/generators/docs_generator.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+      template_string = template.read()
+      template_jinja = Template(template_string)
+      result = template_jinja.render({
+        'platform': platform,
+        'data': data,
+        'colors': colors
+      })
+    with open(os.path.join(parsed_args['output_path'][0], platform.lower() + '-queries' + file_format), 'w') as output:
+      print (result, file=output)
```
Comment (rogeriopeixotocx) [inline .github/generators/docs_generator.py:60]: eof newline missing

File: .github/generators/docs_generator.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+severities = {'High': 'HIGH', 'Medium': 'MEDIUM', 'Low': 'LOW', 'Info': 'INFO'}
+colors = {'High': '#C00', 'Medium': '#C60', 'Low': '#CC0', 'Info': '#00C'}
+template_dict = {}
+platforms = []
+
+arg_parser = argparse.ArgumentParser(description='Create query docs')
+arg_parser.add_argument('-p', nargs=1, type=Path, dest='intput_path', help='Path to read metadata.json')
+arg_parser.add_argument('-o', nargs=1, type=Path, dest='output_path', help='Path to output documentation files')
+arg_parser.add_argument('-t', nargs=1, type=Path, dest='templates_path', help='Path to template files')
+arg_parser.add_argument('-f', nargs='+', type=str, dest='formats', help='Formats to be created documentation, must exists on template folder')
```
Comment (rogeriopeixotocx) [inline .github/generators/docs_generator.py:15]: `Documentation formats to be created, the extension must exists in the template folder`

File: .github/generators/requirements.txt [hunk 8]
```diff
@@ -0,0 +1 @@
+jinja2
```
Comment (rogeriopeixotocx) [inline .github/generators/requirements.txt:1]: `jinja2>=2.10.1`

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]
minor changes requested

File: .github/generators/docs_generator.py (hunk 6 above)
Comment (felipe-avelar) [reply .github/generators/docs_generator.py:60]: If we add eof, codacy creates an issue. :/

Comment (rogeriopeixotocx) [reply .github/generators/docs_generator.py:60]: login to Codacy and mark it as an ignored issue

File: .github/generators/docs_generator.py (hunk 3 above)
Comment (felipe-avelar) [reply .github/generators/docs_generator.py:38]: Not really, descriptionUrl can be string or array of strings. This line checks if it is an string and creates a list with that string alone...

File: docs/css/custom.css [hunk 9]
```diff
@@ -0,0 +1,5 @@
+.table {
+  width: 70vw;
+  table-layout: fixed;
+  word-wrap: break-word;
+}
```
Comment (rogeriopeixotocx) [inline docs/css/custom.css:5]: EOF new line

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]

**Review by rogeriopeixotocx — APPROVED** [review]

**Review by joaoReigota1 — APPROVED** [review]
LGTM

**Review by ruigomescx — APPROVED** [review]

**Pull request merged**